packages/@lightningjs/ui-components/src/components/Icon/Icon.js

```diff
@@ -92,15 +77,17 @@ export default class Icon extends Base {
       this.shader = undefined;
     }
   }
-}
 
-function getIconTemplate(icon, w, h) {
-  const template = { w, h };
-  template.texture = {
-    type: CustomImageTexture,
-    w,
-    h,
-    src: icon
-  };
-  return template;
+  _getIconTemplate() {
+    const { icon, w, h } = this;
+    const template = { w, h };
+    template.texture = {
+      type: CustomImageTexture,
+      w,
+      h,
+      src: icon
+    };
+    template.color = getValidColor(this.color || this.style.color);
+    return template;
```

Comment:
can this whole function be simplified to something like this?
```js
_getIconPatch() {
  return {
    texture: {
      type: CustomImageTexture,
      w: this.w,
      h: this.h,
      src: this.icon
    },
    color: getValidColor(this.color || this.style.color)
  };
}
```

Reply:
Also might want to only apply the color if the color is passed in. I think this might break things for SVGs, just double check those

Reply:
i think the root object would also need w and h? but yeah i think this is a easier to follow too

Reply:
@anthony9187 you're right, just missed it in the first pass, updated the above for clarity

Reply:
Should be handled now. Good callout